cobc/typeck.c

```diff
@@ -5772,41 +5785,40 @@ expr_expand (cb_tree *x)
 			*x = p->x;
 			goto start;
 		}
-		expr_expand (&p->x);
+		build_expr_expand (&p->x);
 		if (p->y) {
-			expr_expand (&p->y);
+			build_expr_expand (&p->y);
 		}
 	}
 }
 
 static cb_tree
-cb_expr_finish (void)
+build_expr_finish (void)
 {
-	/* Reduce all */
-	(void)expr_reduce (0);
+	/* Reduce all (prio of token 0 is smaller than all other ones) */
+	(void)build_expr_reduce (0);
 
-	if (!expr_stack[3].value) {
-		/* TODO: Add test case for this to syn_misc.at invalid expression */
-		cb_error (_("invalid expression"));
+	expr_stack[TOPSTACK].value->source_file = cb_source_file;
+	expr_stack[TOPSTACK].value->source_line = cb_exp_line;
+
+	if (expr_index != TOPSTACK+1) {
```

Comment:
currently this PR also includes the _changes_ to that, not only the rename + comment; when it is actually reduced I think we can pull this in _very_ fast

Reply:
I am iterating from the PR to remove changes gradually. Indeed, I was planning to keep some of the changes in the strings, but it does change the testsuite, so I am going to remove them.